aten/src/ATen/native/CPUBlas.cpp

```diff
@@ -822,4 +822,182 @@ void copy(int64_t n, const c10::complex<float> *x, int64_t incx, c10::complex<fl
       n, x, incx, y, incy);
 }
 
-}  // namespace at::native::cpublas
+void brgemm(
+    int64_t M,
+    int64_t N,
+    int64_t K,
+    int64_t bs,
+    int64_t ld_a,
+    int64_t ld_b,
+    int64_t ld_c,
+    const float alpha,
+    const float beta,
+    const at::Half* A,
+    const at::Half* B,
+    const std::vector<std::pair<int64_t, int64_t>>& offsets,
+    float* C) {
+#if AT_MKLDNN_ENABLED() && (defined(__x86_64__) || (defined(_M_X64) && !defined(_M_ARM64EC)))
+  if (Brgemm::device_check(ScalarType::Half)) {
+    Brgemm::call<at::Half, at::Half, float>(
+      M, N, K, bs, ld_a, ld_b, ld_c, alpha, beta, A, B, offsets, C);
+    return;
+  }
+#endif
+  for (int64_t i = 0; i < bs; i++) {
+    gemm(
+        TransposeType::NoTranspose,
+        TransposeType::NoTranspose,
+        N,
+        M,
+        K,
+        alpha,
+        B + offsets[i].second / sizeof(at::Half),
+        ld_b,
+        A + offsets[i].first / sizeof(at::Half),
+        ld_a,
+        i == 0 ? beta : 1.f,
+        C,
+        ld_c);
+  }
+}
+
+void brgemm(
+    int64_t M,
+    int64_t N,
+    int64_t K,
+    int64_t bs,
+    int64_t ld_a,
+    int64_t ld_b,
+    int64_t ld_c,
+    const float alpha,
+    const float beta,
+    const at::BFloat16* A,
+    const at::BFloat16* B,
+    const std::vector<std::pair<int64_t, int64_t>>& offsets,
+    float* C) {
+#if AT_MKLDNN_ENABLED() && (defined(__x86_64__) || (defined(_M_X64) && !defined(_M_ARM64EC)))
+  if (Brgemm::device_check(ScalarType::BFloat16)) {
+    Brgemm::call<at::BFloat16, at::BFloat16, float>(
+      M, N, K, bs, ld_a, ld_b, ld_c, alpha, beta, A, B, offsets, C);
+    return;
+  }
+#endif
+  for (int64_t i = 0; i < bs; i++) {
+    gemm(
+        TransposeType::NoTranspose,
+        TransposeType::NoTranspose,
+        N,
+        M,
+        K,
+        alpha,
+        B + offsets[i].second / sizeof(at::BFloat16),
+        ld_b,
+        A + offsets[i].first / sizeof(at::BFloat16),
+        ld_a,
+        i == 0 ? beta : 1.f,
+        C,
+        ld_c);
+  }
+}
+
+void brgemm(
+    int64_t M,
+    int64_t N,
+    int64_t K,
+    int64_t ld_a,
+    int64_t ld_b,
+    int64_t ld_c,
+    const float alpha,
+    const float beta,
+    const at::Half* A,
+    const at::Half* B,
+    float* C) {
```

Comment:
perhaps it is simpler to make "bs" as an optional arg defaulting to "1" so that you don't have to create multiple function defs?

Reply:
Removed the versions with batch size since they are not used yet.